### ISSUE TITLE
Object of class Phalcon\Db\RawValue could not be converted to float

### ISSUE THREAD

**Issue by mzf** [phalcon/cphalcon#12766]
### Expected and Actual Behavior

> **Describe what you are trying to achieve and what goes wrong.**

> Provide output if related. Provide coredump if any. Use https://github.com/phalcon/cphalcon/wiki/Generating-a-backtrace as reference

```php
<?php
namespace Tasks;

use PhalconExt\Cli\Task;

class LogsTasks extends \Phalcon\Mvc\Model
{

    /**
     *
     * @var integer
     * @Primary
     * @Identity
     * @Column(type="integer", length=10, nullable=false)
     */
    public $id;

    /**
     *
     * @var double
     * @Column(type="double", length=9, nullable=true)
     */
    public $exec_time;

    /**
     *
     * @var integer
     * @Column(type="integer", length=10, nullable=false)
     */
    public $pid;

}

class IssueTask extends Task
{

    /**
     * @help("Воспроизведение ошибки")
     * @paramsFormat("<y>cli.php {task} {action}</y>")
     * @shortcut("12766")
     * @param type $param
     */
    public function issue12766Action($parameters = null)
    {
        $model = new LogsTasks;
        $model->pid = 1212;
        $model->save();

        $model->exec_time = 0.1212;

        $saved = $model->update();
    }
}

```
DB Scheme
```sql
CREATE TABLE `logs_tasks` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,  
  `exec_time` decimal(9,3) DEFAULT NULL,
  `pid` int(10) unsigned NOT NULL,
  PRIMARY KEY (`id`),
  KEY `pid` (`pid`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8;
```
When update model get error
```
Notice: Object of class Phalcon\Db\RawValue could not be converted to float in Z:\OpenServer\server\trunk\app\cli\tasks\IssueTask.php on line 56
```

### Details

* Phalcon version: 3.1.1
* PHP Version:  PHP 7.0.15 (cli) (built: Jan 17 2017 13:44:27) ( NTS )
* Operating System: Win 7
* Database: MariaDB-5.5-x64

**Comment by sergeyklay** [comment]
@Jurigag Could you please take a look

**Comment by Jurigag** [comment]
Yea, can you provide your table schema?

**Comment by mzf** [comment]
@Jurigag added in start post

**Comment by 409H** [comment]
What happens if you do `$model->exec_time = floatval(0.1212);`

**Comment by mzf** [comment]
@409H same error on $this->update() line

**Comment by Jurigag** [comment]
Can't reproduce.

**Comment by mzf** [comment]
![Screenshot](https://i.gyazo.com/9dff2ae915b41ef51f9b4e48cc9b2e3e.png)

**Comment by mzf** [comment]
I test this code on Phalcon 3.0.4 work fine.
On 3.1.0 same error

**Comment by Jurigag** [comment]
I do't know, you sure you posted full code of your model?

**Comment by mzf** [comment]
@Jurigag i ran you code and get error
```
Fatal error: Uncaught Error: Class 'FactoryDefault' not found
```

**Comment by Jurigag** [comment]
Oh i forgot use statements:

```
use Phalcon\Di\FactoryDefault;
use Phalcon\Mvc\Model;
```

**Comment by mzf** [comment]
Your example work fine on Phalcon 3.1.1. Thanks!
I'll figure out what I've done

**Comment by Jurigag** [comment]
Maybe you need to clear metadata? I don't know really.

**Comment by mzf** [comment]
Try this example please. 
```
<?
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

use Phalcon\Di\FactoryDefault;
use Phalcon\Mvc\Model;

class ModelDynamicUpdate extends \Phalcon\Mvc\User\Plugin
{

    /**
     * @var boolean
     */
    protected $dynamicUpdate;

    /**
     * @param boolean $dynamicUpdate
     */
    public function __construct($dynamicUpdate = true)
    {
        $this->dynamicUpdate = $dynamicUpdate;
    }

    /**
     * @param \Phalcon\Events\Event               $event
     * @param \Phalcon\Mvc\Model\ManagerInterface $modelsManager
     * @param \Phalcon\Mvc\ModelInterface         $model
     */
    public function afterInitialize(\Phalcon\Events\Event $event, \Phalcon\Mvc\Model\ManagerInterface $modelsManager, \Phalcon\Mvc\ModelInterface $model)
    {
        $modelsManager->useDynamicUpdate($model, $this->dynamicUpdate);
    }
}

$di = new FactoryDefault();
$di->set(
    'db', function () {
    $adapter = new \Phalcon\Db\Adapter\Pdo\Mysql(
        [
        'host' => 'localhost',
        'username' => 'root',
        'password' => '',
        'dbname' => 'seomixr_server',
        'options' => [
            PDO::ATTR_EMULATE_PREPARES => false,
            PDO::ATTR_STRINGIFY_FETCHES => false,
            PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8',
        ],
        'charset' => 'utf8',
        ]
    );

    return $adapter;
}
);

$di->set('modelsManager', function () {

    $eventsManager = new \Phalcon\Events\Manager;
    $modelsManager = new Model\Manager;
    $eventsManager->attach("modelsManager", new ModelDynamicUpdate(true));
    $modelsManager->setEventsManager($eventsManager);

    return $modelsManager;
});



require "LogsTasks.php";

$model = new LogsTasks;
$model->pid = 1212;
$model->save();

$model->exec_time = 0.1212;

$saved = $model->update();

var_dump($saved); // true

?>
```

**Comment by Jurigag** [comment]
Yea, now i see, let me fix it.

**Comment by Jurigag** [comment]
Created PR.

**Comment by sergeyklay** [comment]
Fixed in the `3.1.x` branch.
